agent-control/src/cli/on_host/config_gen.rs

```diff
@@ -1,14 +1,311 @@
+    #[arg(long)]
+    auth_parent_client_secret: String,
+
+    /// Auth token corresponding to the parent system identity.
+    #[arg(long)]
+    auth_parent_token: String,
+
+    /// When (`auth_token` or `auth_client_id` + `auth_client_secret`) are set, this path is used
+    /// to store the identity key. Otherwise, the path is expected to contain the already provided
+    /// private key was already provided.
+    #[arg(long)]
+    auth_private_key_path: PathBuf,
+
+    /// Client identifier corresponding to an already provisioned identity. No identity creation is performed,
+    /// therefore setting this up also requires an existing private key pointed in `auth_private_key_path`.
+    #[arg(long)]
+    auth_client_id: String,
+
+    /// Proxy configuration
+    #[command(flatten)]
+    proxy_config: Option<ProxyConfig>,
 }
 
 /// Generates the Agent Control configuration and any requisite according to the provided inputs.
-pub fn generate_config(_inputs: ConfigInputs) -> Result<(), CliError> {
-    unimplemented!("TODO: implement config generation")
+pub fn generate_config(args: Args) -> Result<(), CliError> {
+    info!("Generating Agent Control configuration");
+    let yaml = gen_config(&args, provide_identity)?;
+
+    std::fs::write(&args.output_path, yaml).map_err(|err| {
+        CliError::Command(format!(
+            "error writing the configuration file to '{}': {}",
+            args.output_path.to_string_lossy(),
+            err
+        ))
+    })?;
+    info!(config_path=%args.output_path.to_string_lossy(), "Agent Control configuration generated successfully");
+    Ok(())
+}
+
+/// Generates the configuration according to args using the provided function to generate the identity.
+fn gen_config<F>(args: &Args, provide_identity_fn: F) -> Result<String, CliError>
+where
+    F: Fn(&Args) -> Result<Identity, CliError>,
+{
+    let fleet_control = if !args.fleet_enabled {
+        None
+    } else {
+        let Identity {
+            client_id,
+            private_key_path,
+        } = provide_identity_fn(args)?;
+
+        Some(FleetControl {
+            endpoint: args.region.opamp_endpoint().to_string(),
+            signature_validation: SignatureValidation {
+                public_key_server_url: args.region.public_key_endpoint().to_string(),
+            },
+            fleet_id: args.fleet_id.to_string(),
+            auth_config: AuthConfig {
+                token_url: args.region.token_renewal_endpoint().to_string(),
+                client_id,
+                provider: "local".to_string(),
+                private_key_path: private_key_path.to_string_lossy().to_string(),
+            },
+        })
+    };
+
+    let config = Config {
+        fleet_control,
+        server: Server { enabled: true },
+        proxy: args.proxy_config.clone(),
+        agents: args.agent_set.into(),
+    };
+
+    serde_yaml::to_string(&config)
```

Comment:
Total nit but if we are not using the `String` for anything else than writing the file I'd serialize it to binary data `&[u8]` instead.